checkov/cloudformation/checks/resource/aws/KMSKeyWildCardPrincipal.py

```diff
@@ -0,0 +1,61 @@
+from checkov.common.models.enums import CheckResult, CheckCategories
+from checkov.cloudformation.checks.resource.base_resource_value_check import BaseResourceValueCheck
+
+
+def get_recursively(search_dict, field):
+    """
+    Takes a dict with nested lists and dicts,
+    and searches all dicts for a key of the field
+    provided.
+    """
+    fields_found = []
+
+    for key, value in search_dict.items():
+
+        if key == field:
+            fields_found.append(value)
+
+        elif isinstance(value, dict):
+            results = get_recursively(value, field)
+            for result in results:
+                fields_found.append(result)
+
+        elif isinstance(value, list):
+            for item in value:
+                if isinstance(item, dict):
+                    more_results = get_recursively(item, field)
+                    for another_result in more_results:
+                        fields_found.append(another_result)
+
+    return fields_found
+
+
+class KMSKeyWildCardPrincipal(BaseResourceValueCheck):
+    def __init__(self):
+        name = "Ensure KMS key policy does not contain wildcard (*) principal"
+        id = "CKV_AWS_33"
+        supported_resources = ['AWS::KMS::Key']
+        categories = [CheckCategories.ENCRYPTION]
+        super().__init__(name=name, id=id, categories=categories, supported_resources=supported_resources)
+
+    def get_inspected_key(self):
+        return 'Properties/KeyPolicy/Statement/Principal'
+
+    def scan_resource_conf(self, conf):
+        if conf.get('Properties'):
+            if conf['Properties'].get('KeyPolicy'):
+                policy_block = conf['Properties']['KeyPolicy']
+                principals_list = get_recursively(policy_block, 'Principal')
```

Comment:
@avishayil  looks great! 
do you think the recursive function can be replaced with `dpath`? - was not sure since i haven't drilled down a lot on it.

Reply:
Hi, I thought about using `dpath` initially, but I couldn't get the `conf` object to work with it.
For example:
```python
for x in dpath.util.search(str(policy_block), "Statement", yielded=True):
    print(x)
```

ends with `dpath` error: `E       TypeError: __init__() missing 3 required positional arguments: 'x', 'start_mark', and 'end_mark'`

That's why I chose the recursive function to resolve the principals.

Reply:
Makes sense. Thank you!